functions/src/test/mocks.ts

```diff
@@ -239,7 +239,7 @@ export const firestoreMock = {
             return null
         }
     }
-}
+}) as any as FirebaseFirestore.Firestore
```

Comment:
`as any as Firebase...`?? how does that work?

Reply:
This is how to do partial mocking imo, cast _up_ to `any` then immediately cast _down_ to the type. Then the callers don't have to use @ts-ignore (which is ugly)